addons/website/controllers/main.py

```diff
@@ -45,14 +44,14 @@ def index(self, **kw):
         return self.page(page)
 
     #------------------------------------------------------
-    # Login - overwrite of the web login so that regular users are redirected to the backend 
+    # Login - overwrite of the web login so that regular users are redirected to the backend
     # while portal users are redirected to the frontend by default
     #------------------------------------------------------
     @http.route(website=True, auth="public")
     def web_login(self, redirect=None, *args, **kw):
         r = super(Website, self).web_login(redirect=redirect, *args, **kw)
         if not redirect and request.params['login_success']:
-            if request.registry['res.users'].has_group(request.cr, request.uid, 'base.group_user'):
+            if request.env.user.has_group('base.group_user'):
```

Comment:
No, it works as expected
ref https://github.com/odoo/odoo/blob/master/openerp/addons/base/res/res_users.py#L541